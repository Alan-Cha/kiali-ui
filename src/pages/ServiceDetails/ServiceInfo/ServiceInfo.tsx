import * as React from 'react';
import { Link } from 'react-router-dom';
import ServiceId from '../../../types/ServiceId';
import ServiceInfoBadge from './ServiceInfoBadge';
import ServiceInfoCard from './ServiceInfoCard';
import { Endpoints, Pod, Port, Source, Rule } from '../../../types/ServiceInfo';
import * as API from '../../../services/Api';
import { ToastNotification, ToastNotificationList, Col, Row } from 'patternfly-react';

type ServiceInfoState = {
  labels?: Map<string, string>;
  type: string;
  name: string;
  ip: string;
  ports?: Port[];
  endpoints?: Endpoints[];
  pods?: Pod[];
  rules?: Rule[];
  dependencies?: Source[];
  error: boolean;
  errorMessage: string;
};

class ServiceInfo extends React.Component<ServiceId, ServiceInfoState> {
  constructor(props: ServiceId) {
    super(props);
    this.state = {
      labels: new Map(),
      name: '',
      type: '',
      ip: '',
      ports: [],
      pods: [],
      rules: [],
      dependencies: [],
      error: false,
      errorMessage: ''
    };
  }

  componentWillMount() {
    this.fetchServiceDetails(this.props);
  }

  componentWillReceiveProps(nextProps: ServiceId) {
    this.fetchServiceDetails(nextProps);
  }

  fetchServiceDetails(props: ServiceId) {
    console.log('Fetching info of a service...');
    API.GetServiceDetail(props.namespace, props.service)
      .then(response => {
        console.log(response['data']);
        let data = response['data'];
        this.setState({
          labels: data.labels,
          name: data.name,
          type: data.type,
          ports: data.ports,
          endpoints: data.endpoints,
          pods: data.pods,
          dependencies: data.dependencies,
          rules: data.route_rules,
          ip: data.ip
        });
      })
      .catch(error => {
        this.setState({
          error: true,
          errorMessage: 'Could not connect to server'
        });
        console.log(error);
      });
  }

  render() {
    return (
      <div>
        {this.state.error ? (
          <ToastNotificationList>
            <ToastNotification type="danger">
              <span>
                <strong>Error </strong>
                {this.state.errorMessage}
              </span>
            </ToastNotification>
          </ToastNotificationList>
        ) : null}
        <div className="container-fluid container-cards-pf">
          <Row className="row-cards-pf">
            <Col>
              <ServiceInfoCard
                iconType="pf"
                iconName="service"
                title={this.state.name}
                items={
                  <Row>
                    <Col xs={12} sm={6} md={2} lg={2}>
                      <div className="progress-description">
                        <strong>Labels</strong>
                      </div>
                      {Object.keys(this.state.labels || new Map()).map((key, i) => (
                        <ServiceInfoBadge
                          key={'label_' + i}
                          scale={0.8}
                          style="plastic"
                          color="#0088ce"
                          leftText={key}
                          rightText={this.state.labels ? this.state.labels[key] : ''}
                        />
                      ))}
                      <div>
                        <strong>Type</strong> {this.state.type ? this.state.type : ''}
                      </div>
                      <div>
                        <strong> Ip</strong> {this.state.ip ? this.state.ip : ''}
                      </div>
                    </Col>
                    <Col xs={12} sm={6} md={4} lg={4}>
                      <div className="progress-description">
                        <strong>Ports</strong>
                      </div>
                      <ul style={{ listStyleType: 'none' }}>
                      {(this.state.ports || []).map((port, i) => (
                        <li key={'port_' + i}>
                          {port.protocol} {port.name} ({port.port})
                        </li>
                      ))}
                      </ul>
                    </Col>
                    <Col xs={12} sm={6} md={6} lg={6}>
                      <div className="progress-description">
                        <strong>Endpoints</strong>
                      </div>
                      {(this.state.endpoints || []).map((endpoint, i) => (
                        <Row key={'endpoint_' + i}>
                          <Col xs={12} sm={12} md={12} lg={12}>
                            <ul style={{ listStyleType: 'none' }}>
                              {(endpoint.addresses || []).map((address, u) => (
                                <li key={'endpoint_' + i + '_address_' + u}>
                                  <strong>{address.ip} </strong>: {address.name}
                                </li>
                              ))}
                            </ul>
                          </Col>
                        </Row>
                      ))}
                    </Col>
                  </Row>
                }
              />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoCard
                iconType="fa"
                iconName="cube"
                title="Pods"
                items={(this.state.pods || []).map((pod, u) => (
                  <div key={'pods_' + u}>
                    <div>
                      <strong>Pod</strong>: {pod['name']}
                    </div>
                    <ul style={{ listStyleType: 'none' }}>
                      {Object.keys(pod.labels || new Map()).map((key, i) => (
                        <li>
                          <ServiceInfoBadge
                            key={'pod_labels_badge_' + i}
                            scale={0.8}
                            style="plastic"
                            color="green"
                            leftText={key}
                            rightText={pod.labels ? pod.labels[key] : ''}
                          />
                        </li>
                      ))}
                    </ul>
                  </div>
                ))}
              />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoCard
                iconType="pf"
                iconName="route"
                title="Source Services"
                items={Object.keys(this.state.dependencies || new Map()).map((key, u) => (
                  <div key={'dependencies_' + u}>
                    <div className="progress-description">
                      <strong>To: </strong> {key}
                    </div>
                    <ul style={{ listStyleType: 'none' }}>
                      {(this.state.dependencies ? this.state.dependencies[key] : []).map((dependency, i) => {
                        let nVersion = dependency.indexOf('/');
                        let nNamespace = dependency.indexOf('.');
                        let servicename = dependency.substring(0, nNamespace);
                        let namespace = dependency.substring(nNamespace + 1, nVersion);
                        if (servicename.length > 0 && namespace.length > 0) {
                          let to = '/namespaces/' + namespace + '/services/' + servicename;
                          return (
                            <Link key={to} to={to}>
                              <li key={'dependencies_' + u + '_dependency_' + i}>{dependency}</li>
                            </Link>
                          );
                        } else {
                          return <li key={'dependencies_' + u + '_dependency_' + i}>{dependency}</li>;
                        }
                      })}
                    </ul>
                  </div>
                ))}
              />
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <ServiceInfoCard
                iconType="pf"
                iconName="settings"
                title="Istio Route Rules"
                items={(this.state.rules || []).map((rule, i) => (
                  <div key={'rule' + i}>
                    <div>
                      <strong>Name</strong> : {rule.name}
                    </div>
                    <div>
                      <strong>Precendence</strong> : {rule.precedence}
                    </div>
                    <div>
                      <strong>Route</strong>:
                      <ul style={{ listStyleType: 'none' }}>
                        {(rule.route || []).map((label, u) =>
                          Object.keys(label.labels || new Map()).map((key, n) => (
                            <li key={'rule_' + i + '_label_' + u + '_n_' + n}>
                              <ServiceInfoBadge
                                key={'rule_labels_badge_' + i}
                                scale={0.8}
                                style="plastic"
                                color="green"
                                leftText={key}
                                rightText={label.labels ? label.labels[key] : ''}
                              />
                            </li>
                          ))
                        )}
                      </ul>
                    </div>
                    <div>
                      {!rule.match ? null : (
                        <div>
                          <strong>Match</strong>:
                          <textarea className="form-control textarea-resize" readOnly={true}>
                            {JSON.stringify(rule.match, null, 2)}
                          </textarea>
                        </div>
                      )}
                    </div>
                    <hr />
                  </div>
                ))}
              />
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

export default ServiceInfo;
