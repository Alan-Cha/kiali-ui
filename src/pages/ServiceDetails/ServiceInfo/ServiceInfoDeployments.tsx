import * as React from 'react';
import Badge from '../../../components/Badge/Badge';
import { Deployment } from '../../../types/ServiceInfo';
import { Col, Row, Icon } from 'patternfly-react';
import PfInfoCard from '../../../components/Pf/PfInfoCard';

interface ServiceInfoDeploymentsProps {
  deployments?: Deployment[];
}

class ServiceInfoDeployments extends React.Component<ServiceInfoDeploymentsProps> {
  constructor(props: ServiceInfoDeploymentsProps) {
    super(props);
  }

  render() {
    return (
      <PfInfoCard
        iconType="fa"
        iconName="cube"
        title="Deployments"
        items={(this.props.deployments || []).map((deployment, u) => (
          <Row key={'deployments_' + u}>
            <Col xs={12}>
              <div>
                <p>
                  <strong>{deployment.name}</strong>
                </p>
                <div key="labels">
                  {Object.keys(deployment.labels || new Map()).map((key, i) => (
                    <Badge
                      key={'deployment_' + i}
                      scale={0.8}
                      style="plastic"
                      color="green"
                      leftText={key}
                      rightText={deployment.labels ? deployment.labels[key] : ''}
                    />
                  ))}
                </div>
                <div>
                  <strong>Pod status: </strong> {deployment.available_replicas} / {deployment.replicas}{' '}
                  <Icon
                    type="pf"
                    name={deployment.available_replicas < deployment.replicas ? 'warning-triangle-o' : 'ok'}
                  />
                </div>
                {deployment.autoscaler.name !== '' && (
                  <div>
                    <strong>Autoscaler: </strong>
                    from {deployment.autoscaler.min_replicas} to {deployment.autoscaler.max_replicas} pods ({
                      deployment.autoscaler.target_cpu_utilization_percentage
                    }% CPU)
                  </div>
                )}
              </div>
              <hr />
            </Col>
          </Row>
        ))}
      />
    );
  }
}

export default ServiceInfoDeployments;
