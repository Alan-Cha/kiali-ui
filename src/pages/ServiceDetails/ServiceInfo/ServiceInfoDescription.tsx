import * as React from 'react';
import ServiceInfoBadge from './ServiceInfoBadge';
import ServiceInfoCard from './ServiceInfoCard';
import { Col, Row } from 'patternfly-react';
import { Endpoints, Port } from '../../../types/ServiceInfo';

interface ServiceInfoDescriptionProps {
  name?: string;
  labels?: Map<string, string>;
  type?: string;
  ip?: any;
  ports?: Port[];
  endpoints?: Endpoints[];
}

class ServiceInfoDescription extends React.Component<ServiceInfoDescriptionProps> {
  constructor(props: ServiceInfoDescriptionProps) {
    super(props);
  }

  render() {
    return (
      <ServiceInfoCard
        iconType="pf"
        iconName="service"
        title={this.props.name}
        items={
          <Row>
            <Col xs={12} sm={6} md={2} lg={2}>
              <div className="progress-description">
                <strong>Labels</strong>
              </div>
              {Object.keys(this.props.labels || new Map()).map((key, i) => (
                <div key={'label_' + i}>
                  <ServiceInfoBadge
                    scale={0.8}
                    style="plastic"
                    color="#0088ce"
                    leftText={key}
                    rightText={this.props.labels ? this.props.labels[key] : ''}
                  />
                </div>
              ))}
              <div>
                <strong>Type</strong> {this.props.type ? this.props.type : ''}
              </div>
              <div>
                <strong> Ip</strong> {this.props.ip ? this.props.ip : ''}
              </div>
            </Col>
            <Col xs={12} sm={6} md={4} lg={4}>
              <div className="progress-description">
                <strong>Ports</strong>
              </div>
              <ul style={{ listStyleType: 'none' }}>
                {(this.props.ports || []).map((port, i) => (
                  <li key={'port_' + i}>
                    {port.protocol} {port.name} ({port.port})
                  </li>
                ))}
              </ul>
            </Col>
            <Col xs={12} sm={6} md={6} lg={6}>
              <div className="progress-description">
                <strong>Endpoints</strong>
              </div>
              {(this.props.endpoints || []).map((endpoint, i) => (
                <Row key={'endpoint_' + i}>
                  <Col xs={12} sm={12} md={12} lg={12}>
                    <ul style={{ listStyleType: 'none' }}>
                      {(endpoint.addresses || []).map((address, u) => (
                        <li key={'endpoint_' + i + '_address_' + u}>
                          <strong>{address.ip} </strong>: {address.name}
                        </li>
                      ))}
                    </ul>
                  </Col>
                </Row>
              ))}
            </Col>
          </Row>
        }
      />
    );
  }
}

export default ServiceInfoDescription;
