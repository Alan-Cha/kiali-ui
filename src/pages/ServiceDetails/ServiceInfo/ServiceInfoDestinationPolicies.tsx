import * as React from 'react';
import { DestinationPolicy } from '../../../types/ServiceInfo';
import PfInfoCard from '../../../components/Pf/PfInfoCard';
import ServiceInfoBadge from './ServiceInfoBadge';

interface ServiceInfoDestinationPoliciesProps {
  destinationPolicies?: DestinationPolicy[];
}

class ServiceInfoDestinationPolicies extends React.Component<ServiceInfoDestinationPoliciesProps> {
  constructor(props: ServiceInfoDestinationPoliciesProps) {
    super(props);
  }

  render() {
    return (
      <PfInfoCard
        iconType="pf"
        iconName="settings"
        title="Istio Destination Policies"
        items={(this.props.destinationPolicies || []).map((dPolicy, i) => {
          let dPolicySource;
          if (dPolicy.source) {
            dPolicySource = (
              <div>
                <strong>Source</strong>
                {': '}
                {dPolicy.source.name}
              </div>
            );
          }
          let destinationLabels;
          if (dPolicy.destination && dPolicy.destination.labels) {
            let labels = Object.keys(dPolicy.destination.labels).map((key, n) => (
              <li key={'_label_' + key + '_n_' + n}>
                <ServiceInfoBadge
                  scale={0.8}
                  style="plastic"
                  color="green"
                  leftText={key}
                  rightText={dPolicy.destination.labels ? dPolicy.destination.labels[key] : ''}
                />
              </li>
            ));
            destinationLabels = (
              <div>
                <strong>Destination</strong>:
                <ul style={{ listStyleType: 'none' }}>{labels}</ul>
              </div>
            );
          }
          let loadBalancing;
          if (dPolicy.loadbalancing) {
            loadBalancing = (
              <div>
                <strong>LoadBalancing</strong>
                {': '}
                {dPolicy.loadbalancing.name}
              </div>
            );
          }
          let circuitBreaker;
          if (dPolicy.circuitBreaker) {
            circuitBreaker = (
              <div>
                <strong>CircuitBreaker</strong>
                {': simpleCb'}
                <ul style={{ listStyleType: 'none' }}>
                  {!dPolicy.circuitBreaker.simpleCb.maxConnections ? null : (
                    <li>maxConnections: {dPolicy.circuitBreaker.simpleCb.maxConnections}</li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.httpMaxPendingRequests ? null : (
                    <li>httpMaxPendingRequests: {dPolicy.circuitBreaker.simpleCb.httpMaxPendingRequests}</li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.httpMaxRequests ? null : (
                    <li>httpMaxRequests: {dPolicy.circuitBreaker.simpleCb.httpMaxRequests}</li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.sleepWindow ? null : (
                    <li>sleepWindow: {dPolicy.circuitBreaker.simpleCb.sleepWindow}</li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.httpConsecutiveErrors ? null : (
                    <li>httpConsecutiveErrors: {dPolicy.circuitBreaker.simpleCb.httpConsecutiveErrors}</li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.httpDetectionInterval ? null : (
                    <li>httpDetectionInterval: {dPolicy.circuitBreaker.simpleCb.httpDetectionInterval}</li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.httpMaxRequestsPerConnection ? null : (
                    <li>
                      httpMaxRequestsPerConnection: {dPolicy.circuitBreaker.simpleCb.httpMaxRequestsPerConnection}
                    </li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.httpMaxEjectionPercent ? null : (
                    <li>httpMaxEjectionPercent: {dPolicy.circuitBreaker.simpleCb.httpMaxEjectionPercent}</li>
                  )}
                  {!dPolicy.circuitBreaker.simpleCb.httpMaxRetries ? null : (
                    <li>httpMaxRetries: {dPolicy.circuitBreaker.simpleCb.httpMaxRetries}</li>
                  )}
                </ul>
              </div>
            );
          }
          return (
            <div key={'rule' + i}>
              <div>
                <strong>Name</strong>
                {': '}
                {dPolicy.name}
              </div>
              {dPolicySource}
              {destinationLabels}
              {loadBalancing}
              {circuitBreaker}
              <hr />
            </div>
          );
        })}
      />
    );
  }
}

export default ServiceInfoDestinationPolicies;
