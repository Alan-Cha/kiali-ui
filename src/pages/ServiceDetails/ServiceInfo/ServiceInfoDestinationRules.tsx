import * as React from 'react';
import { DestinationRule } from '../../../types/ServiceInfo';
import { Col, Row } from 'patternfly-react';
import LocalTime from '../../../components/Time/LocalTime';
import DetailObject from '../../../components/Details/DetailObject';

interface ServiceInfoDestinationRulesProps {
  destinationRules?: DestinationRule[];
}

class ServiceInfoDestinationRules extends React.Component<ServiceInfoDestinationRulesProps> {
  constructor(props: ServiceInfoDestinationRulesProps) {
    super(props);
  }

  render() {
    return (
      <div className="card-pf">
        <Row className="row-cards-pf">
          <Col xs={12} sm={12} md={12} lg={12}>
            {(this.props.destinationRules || []).map((destinationRule, i) => (
              <div className="card-pf-body" key={'virtualService' + i}>
                <div>
                  <strong>Name</strong>: {destinationRule.name}
                </div>
                <div>
                  <strong>Created at</strong>: <LocalTime time={destinationRule.created_at} />
                </div>
                <div>
                  <strong>Resource Version</strong>: {destinationRule.resource_version}
                </div>
                {destinationRule.trafficPolicy ? (
                  <DetailObject name="Traffic Policy" detail={destinationRule.trafficPolicy} />
                ) : (
                  undefined
                )}
                {destinationRule.subsets && destinationRule.subsets.length > 0 ? (
                  <DetailObject name="Subsets" detail={destinationRule.subsets} labels={['labels']} />
                ) : (
                  undefined
                )}
                <hr />
              </div>
            ))}
          </Col>
        </Row>
      </div>
    );
  }
}

export default ServiceInfoDestinationRules;
