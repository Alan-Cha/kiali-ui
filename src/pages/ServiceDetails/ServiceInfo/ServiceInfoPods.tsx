import * as React from 'react';
import ServiceInfoBadge from './ServiceInfoBadge';
import ServiceInfoCard from './ServiceInfoCard';
import { Pod } from '../../../types/ServiceInfo';

interface ServiceInfoPodsProps {
  pods?: Pod[];
}

class ServiceInfoPods extends React.Component<ServiceInfoPodsProps> {
  constructor(props: ServiceInfoPodsProps) {
    super(props);
  }

  render() {
    return (
      <ServiceInfoCard
        iconType="fa"
        iconName="cube"
        title="Pods"
        items={(this.props.pods || []).map((pod, u) => (
          <div key={'pods_' + u}>
            <div>
              <strong>Pod</strong>: {pod['name']}
            </div>
            <ul style={{ listStyleType: 'none' }}>
              {Object.keys(pod.labels || new Map()).map((key, i) => (
                <li key={'pod_labels_badge_' + i}>
                  <ServiceInfoBadge
                    scale={0.8}
                    style="plastic"
                    color="green"
                    leftText={key}
                    rightText={pod.labels ? pod.labels[key] : ''}
                  />
                </li>
              ))}
            </ul>
          </div>
        ))}
      />
    );
  }
}

export default ServiceInfoPods;
