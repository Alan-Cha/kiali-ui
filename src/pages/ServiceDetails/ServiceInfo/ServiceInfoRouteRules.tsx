import * as React from 'react';
import { RouteRule } from '../../../types/ServiceInfo';
import PfInfoCard from '../../../components/Pf/PfInfoCard';
import RouteRuleMatch from './ServiceInfoRouteRules/RouteRuleMatch';
import RouteRuleRoute from './ServiceInfoRouteRules/RouteRuleRoute';
import RouteRuleRedirect from './ServiceInfoRouteRules/RouteRuleRedirect';
import RouteRuleHTTPTimeout from './ServiceInfoRouteRules/RouteRuleHTTPTimeout';
import RouteRuleHTTPRetry from './ServiceInfoRouteRules/RouteRuleHTTPRetry';
import RouteRuleHTTPFaultInjection from './ServiceInfoRouteRules/RouteRuleHTTPFaultInjection';
import RouteRuleL4FaultInjection from './ServiceInfoRouteRules/RouteRuleL4FaultInjection';
import RouteRuleIstioService from './ServiceInfoRouteRules/RouteRuleIstioService';
import RouteRuleCorsPolicy from './ServiceInfoRouteRules/RouteRuleCorsPolicy';

interface ServiceInfoRouteRulesProps {
  routeRules?: RouteRule[];
}

class ServiceInfoRouteRules extends React.Component<ServiceInfoRouteRulesProps> {
  constructor(props: ServiceInfoRouteRulesProps) {
    super(props);
  }

  render() {
    return (
      <PfInfoCard
        iconType="pf"
        iconName="settings"
        title="Istio Route Rules"
        items={(this.props.routeRules || []).map((rule, i) => (
          <div key={'rule' + i}>
            <div>
              <strong>Name</strong>: {rule.name}
            </div>
            <div>
              <strong>Precedence</strong>: {rule.precedence}
            </div>
            {rule.match ? <RouteRuleMatch match={rule.match} /> : null}
            {rule.route ? <RouteRuleRoute route={rule.route} /> : null}
            {rule.redirect ? <RouteRuleRedirect redirect={rule.redirect} /> : null}
            {rule.websocketUpgrade ? (
              <div>
                <strong>WebSocket</strong>: {rule.websocketUpgrade}
              </div>
            ) : null}
            {rule.httpReqTimeout ? <RouteRuleHTTPTimeout timeout={rule.httpReqTimeout} /> : null}
            {rule.httpReqRetries ? <RouteRuleHTTPRetry httpReqRetries={rule.httpReqRetries} /> : null}
            {rule.httpFault ? <RouteRuleHTTPFaultInjection httpFault={rule.httpFault} /> : null}
            {rule.l4Fault ? <RouteRuleL4FaultInjection l4Fault={rule.l4Fault} /> : null}
            {rule.mirror ? <RouteRuleIstioService name="Mirror" service={rule.mirror} /> : null}
            {rule.corsPolicy ? <RouteRuleCorsPolicy corsPolicy={rule.corsPolicy} /> : null}
            <hr />
          </div>
        ))}
      />
    );
  }
}

export default ServiceInfoRouteRules;
