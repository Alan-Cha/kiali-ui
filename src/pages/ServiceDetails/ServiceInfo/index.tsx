import { default as ServiceInfoDeployments } from './ServiceInfoDeployments';
import { default as ServiceInfoDescription } from './ServiceInfoDescription';
import { default as ServiceInfoRoutes } from './ServiceInfoRoutes';
import { default as ServiceInfoRules } from './ServiceInfoRules';
import { default as ServiceInfoBadge } from './ServiceInfoBadge';

export { ServiceInfoDeployments, ServiceInfoDescription, ServiceInfoRoutes, ServiceInfoRules, ServiceInfoBadge };
