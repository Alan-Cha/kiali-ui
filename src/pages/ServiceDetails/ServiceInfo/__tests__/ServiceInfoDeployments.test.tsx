import * as React from 'react';
import { shallow } from 'enzyme';
import ServiceInfoDeployments from '../ServiceInfoDeployments';
import { Deployment } from '../../../../types/ServiceInfo';

const deployments: Deployment[] = [
  {
    name: 'reviews-v2',
    labels: { app: 'reviews', version: 'v2' },
    createdAt: '2018-03-14T10:17:52Z',
    resourceVersion: '1234',
    replicas: 5,
    availableReplicas: 3,
    unavailableReplicas: 2,
    autoscaler: {
      name: 'reviews-v2-autoscaler',
      minReplicas: 2,
      maxReplicas: 10,
      targetCPUUtilizationPercentage: 50
    }
  },
  {
    name: 'reviews-v3',
    labels: { app: 'reviews', version: 'v3' },
    createdAt: '2018-03-14T10:17:52Z',
    resourceVersion: '1234',
    replicas: 2,
    availableReplicas: 2,
    unavailableReplicas: 0,
    autoscaler: {
      name: 'reviews-v2-autoscaler',
      minReplicas: 2,
      maxReplicas: 10,
      targetCPUUtilizationPercentage: 50
    }
  },
  {
    name: 'reviews-v1',
    labels: { app: 'reviews', version: 'v1' },
    createdAt: '2018-03-14T10:17:52Z',
    resourceVersion: '1234',
    replicas: 1,
    availableReplicas: 1,
    unavailableReplicas: 0,
    autoscaler: {
      name: 'reviews-v2-autoscaler',
      minReplicas: 2,
      maxReplicas: 10,
      targetCPUUtilizationPercentage: 50
    }
  }
];

describe('#ServiceInfoDeployments render correctly with data', () => {
  it('should render service pods', () => {
    const wrapper = shallow(<ServiceInfoDeployments deployments={deployments} />);
    expect(wrapper).toBeDefined();
    expect(wrapper).toMatchSnapshot();
  });
});
