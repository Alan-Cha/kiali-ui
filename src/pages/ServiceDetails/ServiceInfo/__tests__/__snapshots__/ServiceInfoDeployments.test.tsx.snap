// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#ServiceInfoDeployments render correctly with data should render service pods 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ServiceInfoDeployments
    deployments={
      Array [
        Object {
          "autoscaler": Object {
            "max_replicas": 10,
            "min_replicas": 2,
            "name": "reviews-v2-autoscaler",
            "target_cpu_utilization_percentage": 50,
          },
          "available_replicas": 3,
          "created_at": "2018-03-14T10:17:52Z",
          "labels": Map {
            "app" => "reviews",
            "version" => "v2",
          },
          "name": "reviews-v2",
          "replicas": 5,
          "unavailable_replicas": 2,
        },
        Object {
          "autoscaler": Object {
            "max_replicas": 10,
            "min_replicas": 2,
            "name": "reviews-v2-autoscaler",
            "target_cpu_utilization_percentage": 50,
          },
          "available_replicas": 2,
          "created_at": "2018-03-14T10:17:52Z",
          "labels": Map {
            "app" => "reviews",
            "version" => "v3",
          },
          "name": "reviews-v3",
          "replicas": 2,
          "unavailable_replicas": 0,
        },
        Object {
          "autoscaler": Object {
            "max_replicas": 10,
            "min_replicas": 2,
            "name": "reviews-v2-autoscaler",
            "target_cpu_utilization_percentage": 50,
          },
          "available_replicas": 1,
          "created_at": "2018-03-14T10:17:52Z",
          "labels": Map {
            "app" => "reviews",
            "version" => "v1",
          },
          "name": "reviews-v1",
          "replicas": 1,
          "unavailable_replicas": 0,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "iconName": "cube",
      "iconType": "fa",
      "items": Array [
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            xs={12}
          >
            <div>
              <p>
                <strong>
                  reviews-v2
                </strong>
              </p>
              <div />
              <div>
                <strong>
                  Pod status: 
                </strong>
                 
                3
                 / 
                5
                 
                <Icon
                  name="warning-triangle-o"
                  style={
                    Object {
                      "fontSize": "12px",
                    }
                  }
                  title=""
                  type="pf"
                />
              </div>
              <div>
                <strong>
                  Autoscaler: 
                </strong>
                from 
                2
                 to 
                10
                 pods (
                50
                % CPU)
              </div>
            </div>
            <hr />
          </Col>
        </Row>,
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            xs={12}
          >
            <div>
              <p>
                <strong>
                  reviews-v3
                </strong>
              </p>
              <div />
              <div>
                <strong>
                  Pod status: 
                </strong>
                 
                2
                 / 
                2
                 
                <Icon
                  name="ok"
                  style={
                    Object {
                      "fontSize": "12px",
                    }
                  }
                  title=""
                  type="pf"
                />
              </div>
              <div>
                <strong>
                  Autoscaler: 
                </strong>
                from 
                2
                 to 
                10
                 pods (
                50
                % CPU)
              </div>
            </div>
            <hr />
          </Col>
        </Row>,
        <Row
          bsClass="row"
          componentClass="div"
        >
          <Col
            bsClass="col"
            componentClass="div"
            xs={12}
          >
            <div>
              <p>
                <strong>
                  reviews-v1
                </strong>
              </p>
              <div />
              <div>
                <strong>
                  Pod status: 
                </strong>
                 
                1
                 / 
                1
                 
                <Icon
                  name="ok"
                  style={
                    Object {
                      "fontSize": "12px",
                    }
                  }
                  title=""
                  type="pf"
                />
              </div>
              <div>
                <strong>
                  Autoscaler: 
                </strong>
                from 
                2
                 to 
                10
                 pods (
                50
                % CPU)
              </div>
            </div>
            <hr />
          </Col>
        </Row>,
      ],
      "title": "Deployments",
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "iconName": "cube",
        "iconType": "fa",
        "items": Array [
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Col
              bsClass="col"
              componentClass="div"
              xs={12}
            >
              <div>
                <p>
                  <strong>
                    reviews-v2
                  </strong>
                </p>
                <div />
                <div>
                  <strong>
                    Pod status: 
                  </strong>
                   
                  3
                   / 
                  5
                   
                  <Icon
                    name="warning-triangle-o"
                    style={
                      Object {
                        "fontSize": "12px",
                      }
                    }
                    title=""
                    type="pf"
                  />
                </div>
                <div>
                  <strong>
                    Autoscaler: 
                  </strong>
                  from 
                  2
                   to 
                  10
                   pods (
                  50
                  % CPU)
                </div>
              </div>
              <hr />
            </Col>
          </Row>,
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Col
              bsClass="col"
              componentClass="div"
              xs={12}
            >
              <div>
                <p>
                  <strong>
                    reviews-v3
                  </strong>
                </p>
                <div />
                <div>
                  <strong>
                    Pod status: 
                  </strong>
                   
                  2
                   / 
                  2
                   
                  <Icon
                    name="ok"
                    style={
                      Object {
                        "fontSize": "12px",
                      }
                    }
                    title=""
                    type="pf"
                  />
                </div>
                <div>
                  <strong>
                    Autoscaler: 
                  </strong>
                  from 
                  2
                   to 
                  10
                   pods (
                  50
                  % CPU)
                </div>
              </div>
              <hr />
            </Col>
          </Row>,
          <Row
            bsClass="row"
            componentClass="div"
          >
            <Col
              bsClass="col"
              componentClass="div"
              xs={12}
            >
              <div>
                <p>
                  <strong>
                    reviews-v1
                  </strong>
                </p>
                <div />
                <div>
                  <strong>
                    Pod status: 
                  </strong>
                   
                  1
                   / 
                  1
                   
                  <Icon
                    name="ok"
                    style={
                      Object {
                        "fontSize": "12px",
                      }
                    }
                    title=""
                    type="pf"
                  />
                </div>
                <div>
                  <strong>
                    Autoscaler: 
                  </strong>
                  from 
                  2
                   to 
                  10
                   pods (
                  50
                  % CPU)
                </div>
              </div>
              <hr />
            </Col>
          </Row>,
        ],
        "title": "Deployments",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
